commands/cluster-nodes.md

```diff
@@ -28,26 +28,53 @@ of output:
-8. `link-state`: The state of the link used for the node-to-node cluster bus. We use this link to communicate with the node. Can be `connected` or `disconnected`.
-9. `slot`: A hash slot number or range. Starting from argument number 9, but there may be up to 16384 entries in total (limit never reached). This is the list of hash slots served by this node. If the entry is just a number, is parsed as such. If it is a range, it is in the form `start-end`, and means that the node is responsible for all the hash slots from `start` to `end` including the start and end values.
-
-Meaning of the flags (field number 3):
+3. `hostname`: A human readable string that can be configured via the
+   `cluster-annouce-hostname` setting. The max length of the string is 256
+   characters, excluding the null terminator. The name can contain ASCII alphanumeric characters,
+   '-', and '.' only.
+4. `[,auxiliary_field=value]*`: A list of comma-separated key-value pairs that
+   represent various node properties, such as `shard-id`. There is no intrinsic
+   order among auxiliary fields. The auxiliary fields can appear at
+   different position in the list from release to release. Both the key name and
+   value can contain alphanumeric characters and the characters in
+   `!#$%&()*+-.:;<>?@[]^_{|}~` only. Auxiliary fields are explaned in detail in
+   the section below.
+5. `flags`: A list of comma separated flags: `myself`, `master`, `slave`, `fail?`, `fail`, `handshake`, `noaddr`, `nofailover`, `noflags`. Flags are explained in detail in the next section.
+6. `master`: If the node is a replica, and the master is known, the master node ID, otherwise the "-" character.
+7. `ping-sent`: Milliseconds unix time at which the currently active ping was sent, or zero if there are no pending pings.
+8. `pong-recv`: Milliseconds unix time the last pong was received.
+9. `config-epoch`: The configuration epoch (or version) of the current node (or of the current master if the node is a replica). Each time there is a failover, a new, unique, monotonically increasing configuration epoch is created. If multiple nodes claim to serve the same hash slots, the one with higher configuration epoch wins.
+10. `link-state`: The state of the link used for the node-to-node cluster bus. We use this link to communicate with the node. Can be `connected` or `disconnected`.
+11. `slot`: A hash slot number or range. Starting from argument number 9, but there may be up to 16384 entries in total (limit never reached). This is the list of hash slots served by this node. If the entry is just a number, is parsed as such. If it is a range, it is in the form `start-end`, and means that the node is responsible for all the hash slots from `start` to `end` including the start and end values.
+
+Auxiliary fields (filed number 4):
+* `shard-id`: a 40 characters random string generated when a node is created. A
```

Comment:
```suggestion
* `shard-id`: a 40-random-character string generated when a node is created. A node's shard id changes only when the node joins a different shard via `cluster replicate` and there the node's shard id is updated to its primary's.
```

Reply:
May also not be random in the future. So let's just say 40 character string.

Reply:
Good point. I think the real important property is "globally unique" so added it as the qualifier.